cns/cnireconciler/podinfoprovider.go

```diff
@@ -0,0 +1,36 @@
+package cnireconciler
+
+import (
+	"github.com/Azure/azure-container-networking/cni/api"
+	"github.com/Azure/azure-container-networking/cni/client"
+	"github.com/Azure/azure-container-networking/cns"
+	"k8s.io/utils/exec"
+)
+
+// NewCNIPodInfoProvider returns an implementation of cns.PodInfoByIPProvider
+// that execs out to the CNI and uses the response to build the PodInfo map.
+func NewCNIPodInfoProvider() (cns.PodInfoByIPProvider, error) {
+	cli := client.New(exec.New())
+	state, err := cli.GetEndpointState()
+	if err != nil {
+		return nil, err
```

Comment:
Can we return fmt.ErrorF and append the error with more info say "Failed to invoke CNI.GetEndpointsate(), error: %s", err?